### PR TITLE
[aot] Don't error out if a method fails to JIT in full-aot mode. This…

### PR DESCRIPTION
**Pull request by vargaz**
… was a regression caused by 9302c31f81abd2f5bbf9a8ddd27993e1ece55db2. Fixes #59832.

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -421,7 +421,7 @@ aot_printerrf (MonoAotCompile *acfg, const gchar *format, ...)
 }
 
 static void
-report_loader_error (MonoAotCompile *acfg, MonoError *error, const char *format, ...)
+report_loader_error (MonoAotCompile *acfg, MonoError *error, gboolean fatal, const char *format, ...)
```
Comment (rolfbjarne) [inline mono/mini/aot-compiler.c]: You're never using this new parameter.

Comment (rolfbjarne) [reply mono/mini/aot-compiler.c]: Now you are :smile:

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -439,7 +439,7 @@ report_loader_error (MonoAotCompile *acfg, MonoError *error, const char *format,
 	va_end (args);
 	mono_error_cleanup (error);
 
-	if (acfg->is_full_aot) {
+	if (acfg->is_full_aot && !fatal) {
```
Comment (akoeplinger) [inline mono/mini/aot-compiler.c]: shouldn't this be `&& fatal` ?

**Comment by rolfbjarne** [comment]
This is a blocker for Xamarin.iOS for d15-5, can we get this in (and in all the required branches as well)?

**Comment by vargaz** [comment]
Which branch should we backport to ?

**Comment by rolfbjarne** [comment]
XI/d15-5 tracks 2017-06, so I guess that means 2017-06, 2017-08, 2017-10 (and master).

**Pull request merged**